### PR TITLE
Fix accountsChanged notification

### LINKED ISSUE #1404: Changing accounts not detected unless reloading page on browser
upon changing accounts the first time, it's detected, however, any subsequent account change is not detected unless reloading the page

seen here = http://recordit.co/9n1OWOfaOi

Steps to reproduce: 
- pull latest develop (or grab release build from MM bot) 
- go to uniswap (connect if not already connected) 
- take note of your current account that you are on + balance
- now switch accounts (it works the first time) 
- now switch to a different account (notice how your account nor balance is not updated, but stays on the previous one)

### PR DESCRIPTION
**Pull request by rekmarks**
- Use `json-rpc-engine` notifications to trigger `accountsChanged` events on the inpage provider
  - This currently causes console errors in the dapps, but doesn't actually break anything
  - Removing those errors is probably not straightforward

Fixes #1404

### CODE REVIEW

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -864,6 +882,27 @@ export class BrowserTab extends PureComponent {
 				this.loadUrl();
 			}
 		}
+
+		const numApprovedHosts = Object.keys(approvedHosts).length;
+		const prevNumApprovedHosts = Object.keys(prevApprovedHosts).length;
```
Comment (estebanmino) [inline app/components/Views/BrowserTab/index.js]: `approvedHosts` and `prevApprovedHosts` are always defined?

File: app/core/BackgroundBridge.js [hunk 2]
```diff
@@ -166,19 +164,15 @@ export class BackgroundBridge extends EventEmitter {
...
-		const selectPublicState = ({ isUnlocked, selectedAddress, network }) => {
-			const isEnabled = checkIsEnabled();
-			const isReady = isUnlocked && isEnabled;
+		const selectPublicState = ({ isUnlocked, network }) => {
 			const networkType = Engine.context.NetworkController.state.provider.type;
 			const chainId = Object.keys(NetworkList).indexOf(networkType) > -1 && NetworkList[networkType].chainId;
 			const result = {
 				isUnlocked,
-				isEnabled,
-				selectedAddress: isReady ? selectedAddress.toLowerCase() : null,
```
Comment (estebanmino) [inline app/core/BackgroundBridge.js]: this wasn't necessary?

File: app/components/Views/BrowserTab/index.js (hunk 1 above)
Comment (rekmarks) [reply app/components/Views/BrowserTab/index.js]: Yeah, `approvedHosts` is set in the privacy reducer. It's initial state is `{}`, and the clear action resets it to that.

File: app/core/BackgroundBridge.js (hunk 2 above)
Comment (rekmarks) [reply app/core/BackgroundBridge.js]: It was previously, but no longer. The changes I made match what we did on the extension, and we removed `isEnabled` and `selectedAddress` from the public config store.

Comment (rekmarks) [reply app/core/BackgroundBridge.js]: I should have already removed both tbh, but overlooked it.

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good, QA passed 👍

**Review by estebanmino — APPROVED** [review]
👌 ship it

**Pull request merged**